client/etcd/client.go

```diff
@@ -212,12 +211,13 @@ func fileName(service, zone string) string {
 	return strings.Replace(cacheFileName, string(os.PathSeparator), "_", -1)
 }
 
-func prefix(env string) string {
-	res := kvPrefix
+func fullPrefix(namespace string, env string) string {
+	prefix := concat(keySeparator, concat(namespace, keySeparator))
```

Comment:
nit: it might be readable to just write it as
```
p := namespace
if env != "" {
   p += keySeparator + env
}
return keySeparator + p + keySeparator
```